realm/realm-library/src/syncIntegrationTest/java/io/realm/SyncedRealmTests.java

```diff
@@ -49,6 +49,36 @@
 @RunWith(AndroidJUnit4.class)
 public class SyncedRealmTests extends StandardIntegrationTest {
 
+
+    @Test
+    public void loginLogoutResumeSyncing() throws InterruptedException {
+        String username = UUID.randomUUID().toString();
+        String password = "password";
+        SyncUser user = SyncUser.login(SyncCredentials.usernamePassword(username, password, true), Constants.AUTH_URL);
+
+        SyncConfiguration config = new SyncConfiguration.Builder(user, Constants.USER_REALM)
+                .schema(StringOnly.class)
+                .build();
+
+        Realm realm = Realm.getInstance(config);
+        realm.beginTransaction();
+        realm.createObject(StringOnly.class).setChars("Foo");
+        realm.commitTransaction();
+        SyncManager.getSession(config).uploadAllLocalChanges();
+        user.logout();
+        realm.close();
+
+        user = SyncUser.login(SyncCredentials.usernamePassword(username, password, false), Constants.AUTH_URL);
```

Comment:
You are logging in the same user So the path on disk would be the same. I don't see how this actually test that syncing works again as you just uploaded all changes and nothing is being created on the server in the meantime?